Gems/Atom/RHI/Metal/Code/Source/RHI/CommandListBase.cpp

```diff
@@ -63,9 +63,63 @@ namespace AZ
         {
             if (m_encoder)
             {
+                //Call UseResource on all resources cached for Compute work
+                id<MTLRenderCommandEncoder> renderEncoder = GetEncoder<id<MTLRenderCommandEncoder>>();
+                id<MTLComputeCommandEncoder> computeEncoder = GetEncoder<id<MTLComputeCommandEncoder>>();
+
+                if(m_untrackedResourcesComputeRead.size() > 0)
+                {
+                    AZStd::vector<id <MTLResource>> resourcesToProcessVec(m_untrackedResourcesComputeRead.begin(),
+                                                                          m_untrackedResourcesComputeRead.end());
```

Comment:
why duplicate the vector here? just to make a linear list? 
Does the constructor with 2 iterators guarantee only 1 heap allocation?